src/java/org/apache/cassandra/index/sai/iterators/KeyRangeAntiJoinIterator.java

```diff
@@ -37,7 +37,7 @@ public class KeyRangeAntiJoinIterator extends KeyRangeIterator
 
     private KeyRangeAntiJoinIterator(KeyRangeIterator left, KeyRangeIterator right)
     {
-        super(left.getMinimum(), left.getMaximum(), left.getMaxKeys());
+        super(left.getMinimum(), left.getMaximum(), left.getMaxKeys() - right.getMaxKeys());
```

Comment:
I don't think we can assume all keys would match the right side.
getMaxKeys should return the upper bound on the number of keys returned, so the previous code was good.

Reply:
I found that the left will be 0 for an empty table, and having this count equal to 0 has a special meaning. So it's reverted in my current uncommitted work.